lib/aff4.py

```diff
@@ -2496,13 +2502,20 @@ def _ReadPartial(self, length):
 
     return result
 
-  def Read(self, length):
+  def Read(self, length=None):
     """Read a block of data from the file."""
     result = ""
 
     # The total available size in the file
-    length = int(length)
-    length = min(length, self.size - self.offset)
+    if length is not None:
+        length = int(length)
+        length = min(length, self.size - self.offset)
+    elif self.size < self.max_unbound_read:
```

Comment:
Are you sure this logic is correct? Are we trying to raise here if the read would return more data than is allowed by the max_unbound_read? If so this should be:

length = self.size - self.offset
if lengtn > self.max_unbound_read: raise ....